app/file.go

```diff
@@ -604,3 +604,32 @@ func (a *App) GetFileInfo(fileId string) (*model.FileInfo, *model.AppError) {
 		return result.Data.(*model.FileInfo), nil
 	}
 }
+
+func (a *App) CopyFileInfos(userId string, fileIds []string) ([]string, *model.AppError) {
+	newFileIds := []string{}
+
+	now := time.Now()
+
+	for _, fileId := range fileIds {
+		fileInfo := &model.FileInfo{}
+
+		if result := <-a.Srv.Store.FileInfo().Get(fileId); result.Err != nil {
+			return nil, result.Err
+		} else {
+			fileInfo = result.Data.(*model.FileInfo)
+		}
+
+		fileInfo.Id = model.NewId()
+		fileInfo.CreatorId = userId
+		fileInfo.CreateAt = now.UnixNano() / int64(time.Millisecond)
+		fileInfo.PostId = ""
+
+		if result := <-a.Srv.Store.FileInfo().Save(fileInfo); result.Err != nil {
+			return nil, result.Err
```

Comment:
Maybe we should change how this handle errors? If it fails to copy one row, any rows copied earlier are left behind. I don't think that should be a big deal, but maybe it should return the successfully copied ones or something so the caller can clean them up or something if it errors.

Any thoughts, @crspeller?

Reply:
It's a pretty rare case that it will partially fail, so I think just returning what did copy is fine for most uses.

Reply:
fixed in commit ea326d2